File: Question_LeetcodeAndGeeks/OrderedMap/medium/Q001Leetcode180_hand-of-straights/Q001Leetcode180.java
import java.util.TreeMap;
public class Q001Leetcode180{

    public boolean isNStraightHand(int[] hand, int W) {
         TreeMap<Integer,Integer> tree=new TreeMAp<>();
         for(int& ele: hand) tree.put(val,tree.getOrDefault(val,0)+1);

         while(tree.size()>0){
            int val=tree.fisrtKey();
            for(int count=1;count<=W;count++){
                if(tree.count(val)){
                    tree.put(val,tree.getOrDefault(val,0)-1);
                    if(tree.get(val)==0)tree.remove(val);
                    val++;
                }else return false;
            }
        }
      return true;
         
    }

File: Question_LeetcodeAndGeeks/OrderedMap/medium/Q001Leetcode180_hand-of-straights/Q001Leetcode180.cpp
// #include <bits/stdc++.h>
#include <iostream>
#include <vector>
#include <string>

#define vi vector<int>
#define vii vector<vi>

using namespace std;

auto speedUp = []() {
    std::ios::sync_with_stdio(false);
    cin.tie(nullptr);
    cout.tie(nullptr);
    return nullptr;
}();

bool isNStraightHand(vector<int>& hand, int W) {
        map<int,int> tree;
        for(int& ele: hand) tree[ele]++;
        
        while(tree.size()>0){
            int val=tree.begin()->first;
            for(int count=1;count<=W;count++){
                if(tree.count(val)){
                    tree[val]--;
                    if(tree[val]==0)tree.erase(val);
                    val++;
                }else return false;
            }
        }
      return true;
    }